netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -86,6 +90,19 @@ public NettyClientHandler(Http2ConnectionEncoder encoder, Http2Connection connec
     connection.remote().maxActiveStreams(0);
     connection.local().allowPushTo(false);
     nextStreamId = connection.local().nextStreamId();
+    connection.addListener(new Http2ConnectionAdapter() {
+      @Override
+      public void onGoAwaySent(int lastStreamId, long errorCode, ByteBuf debugData) {
```

Comment:
Why do we need to be aware of go away is sent? Shouldn't we have been the one to send it and so we won't create any more streams?